bika/lims/controlpanel/bika_sampletypes.py

```diff
@@ -5,216 +5,251 @@
+from Products.CMFPlone.utils import safe_unicode
+from zope.interface.declarations import implements
+
+
+# TODO: Separate content and view into own modules!
+
 
 class SampleTypesView(BikaListingView):
     implements(IFolderContentsView, IViewView)
 
     def __init__(self, context, request):
         super(SampleTypesView, self).__init__(context, request)
-        self.catalog = 'bika_setup_catalog'
-        self.contentFilter = {'portal_type': 'SampleType',
-                              'sort_on': 'sortable_title'}
-        self.context_actions = {_('Add'):
-                                {'url': 'createObject?type_name=SampleType',
-                                 'icon': '++resource++bika.lims.images/add.png'}}
+
+        self.catalog = "bika_setup_catalog"
+
+        self.contentFilter = {
+            "portal_type": "SampleType",
+            "sort_on": "sortable_title",
+            "sort_order": "ascending",
+        }
+
+        self.context_actions = {
+            _("Add"): {
+                "url": "createObject?type_name=SampleType",
+                "permission": "Add portal content",
+                "icon": "++resource++bika.lims.images/add.png"}
+        }
+
         self.title = self.context.translate(_("Sample Types"))
-        self.icon = self.portal_url + "/++resource++bika.lims.images/sampletype_big.png"
+        self.icon = "{}/{}".format(
+            self.portal_url,
+            "/++resource++bika.lims.images/sampletype_big.png"
+        )
+
         self.description = ""
         self.show_sort_column = False
         self.show_select_row = False
         self.show_select_column = True
         self.pagesize = 25
 
-        self.columns = {
-            'Title': {'title': _('Sample Type'),
-                      'index': 'sortable_title',},
-            'Description': {'title': _('Description'),
-                            'index': 'description',
-                            'toggle': True},
-            'getHazardous': {'title': _('Hazardous'),
-                             'toggle': True},
-            'getPrefix': {'title': _('Prefix'),
-                          'toggle': True},
-            'getMinimumVolume': {'title': _('Minimum Volume'),
-                                 'toggle': True},
-            'RetentionPeriod': {'title': _('Retention Period'),
-                             'toggle': True},
-            'SampleMatrix': {'title': _('SampleMatrix'),
-                             'toggle': True},
-            'ContainerType': {'title': _('Default Container'),
-                             'toggle': True},
-            'getSamplePoints': {'title': _('Sample Points'),
-                             'toggle': True},
-        }
+        self.columns = collections.OrderedDict((
+            ("Title", {
+                "title": _("Sample Type"),
+                "index": "sortable_title"}),
+            ("Description", {
+                "title": _("Description"),
+                "index": "description",
```

Comment:
The name of the index must be "Description" (first in uppercase).... I missed this one